### PR TITLE
Stats: Add Post Likes to Post Stats Summary Page

### PR DESCRIPTION
**Pull request by youknowriad**
Part of #10348

In this PR, I'm adding a new component to the Post Stats Summary Page. This component shows the Gravatars of the post likes and the total.

screen shot 2017-01-02 at 16 37 14

Questions:

 * What if we have a huge number of likes, should we limit the Gravatars to something like the 100 first likes.
 * I used a `toggleInfo` HoC as a replacement to the old `toggle` mixin used on other stats components. The mixin used to store its state on local storage using the npm library `store`. Should I persist this state. I personally don't think it's worth it, and even if it is, I don't think we should add this to the global state tree, maybe we just keep a behaviour similar to the mixin (calling localstorage directly on the component).

Testing instructions

 - try opening the stats summary page for different posts and see how the new component behaves.
`/stats/post/$postId/$siteId`

### CODE REVIEW

**Comment by youknowriad** [comment]
This is not blocked anymore, so maybe time for a design/code review

**Comment by folletto** [comment]
Heya! I gave it a go:

1. One of the posts I tried showed the header but no likes, and not even the message "This post has not likes yet!" — but... it's the homepage of the site (post ID = 0) :D 
2. I'd remove the panel entirely when there are no likes. (either that, or a different text: "There are no likes on this post yet.")
3. Each avatar has to link somewhere. Either their primary site, or something else more relevant.
4. Can we move the likes card right below the graph? Thoughts?


> What if we have a huge number of likes, should we limit the Gravatars to something like the 100 first likes.

Can we do instead the 100 latest likes which is more useful to see the new people? Also we need to show a placeholder at the end saying "+93" to indicate the list isn't complete (ideally if clicked should still show the full list, if feasible).

**Comment by youknowriad** [comment]
> I'd remove the panel entirely when there are no likes. (either that, or a different text: "There are no likes on this post yet.")

@folletto I would prefer fixing the text, because we have to trigger a network request to see if there are likes or not, and during the "requesting" time, we show up a "loading" indicator. If we decide to remove the panel when the loading ends, the user would notice this change and its not the best experience possible (same if we decide to not show the loading indicator and the panel shows up when the request ends) 

> Can we do instead the 100 latest likes which is more useful to see the new people? Also we need to show a placeholder at the end saying "+93" to indicate the list isn't complete (ideally if clicked should still show the full list, if feasible).

As @timmyc noted on #10361 The endpoint only returns the 90 first likes with avatars. Thus, we won't be able to show the full list.

Time to get my hands dirty and address the feedback tks @folletto

**Comment by karmatosed** [comment]
I would also second some of what @folletto says. I would like us to consider how we can invisible search (if possible). I'm concerned about the user experience of a loader that doesn't return anything.

If we do only show the 90 first likes I'd like us to find a way of showing more or linking to a place that can with a link. Ideally just having them able to load would be good, but I'm getting the impression that's a limitation.

**Comment by timmyc** [comment]
> but I'm getting the impression that's a limitation

At this time the 90 limit on gravatar/likees is a limitation.  We don't have pagination setup for this in the API, and also I don't think the underlying logic to grab the likes supports pagination either.

File: client/my-sites/stats/stats-post-likes/index.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import StatsModuleContent from '../stats-module/content-text';
+import QueryPostLikes from 'components/data/query-post-likes';
+import StatsModulePlaceholder from '../stats-module/placeholder';
+import toggleInfo from '../toggle-info';
+import { isRequestingPostLikes, getPostLikes, countPostLikes } from 'state/selectors';
+
+export const PostLikes = ( { countLikes, isRequesting, likes, opened, postId, siteId, toggle, translate } ) => {
+	const infoIcon = opened ? 'info' : 'info-outline';
+	const classes = {
+		'is-showing-info': opened,
```
Comment (timmyc) [inline client/my-sites/stats/stats-post-likes/index.jsx]: I noticed while the components are loading the `module-header-title` and `infoIcon` are both displayed in this new component.  If we add in a `is-loading` class while the query is still loading then this component will match the behavior of the others on the page.

**Comment by timmyc** [comment]
One note on the missing loading indicator on the title of the component - also could we add analytics tracking for when a gravatar is clicked on?

Also not sure if we wanted to add this in as @folletto requested:

> Also we need to show a placeholder at the end saying "+93"

For grins, this is how the like block looks on a post:

wordpress_com_in_2016__a_year_in review_ _the_wordpress_com_blog

Gravatars in that context are linked to the gravatar site ( i.e. http://en.gravatar.com/timmydcrawford ) - should we do the same here or stick with the user's default site?

**Comment by youknowriad** [comment]
thanks for the review @timmyc 

PR updated in d8ba8c9 to address these notes. I added a placeholder with the remaining likes count, but it needs a design review. I hope this addresses the API limitation issue.

screen shot 2017-01-11 at 10 35 50

**Comment by karmatosed** [comment]
Thanks for the PR update, looking at the screenshot above I can see the + number has some padding issues from size, if you compare the the post links one. I feel they should be the same in format.

I am reluctant to do a full design review sign off until I see this for myself. I have tried to both use mattibot and also /stats/post/$postId/$siteId. On using that though it just refreshed to the blog's insights. I tried on updateomattic. I was wondering if I'm missing a step in trying to recreate to see this PR in action?

**Comment by karmatosed** [comment]
@youknowriad thanks for your help in getting this working. On reviewing a few things:

- Information text should say 'This panel shows the list of people who like your post'. It currently says likes your post.
- The +number at end I do think should be made the same size as the post likes number bubble.

2017-01-11 at 12 20

- With the information block exposed, there seems to be a padding issue to the top:

2017-01-11 at 12 19

One thing worth noting that I think we all know is a expected and a result of a limitation, I really wanted to click the +number. I found myself doing this several times. If there is no way of loading the rest, this is something we I think have to accept as without the number it looks even odder. That said, it is something I think we should review at some point. 

As far as linking to Gravatar this made me wonder if linking to their default blog wouldn't be better. I feel that could be more relevant and more encouraging of engagement than Gravtar.

**Comment by youknowriad** [comment]
Thanks @karmatosed for the review, I updated the component in 98f6ebb. Here is the current design

screen shot 2017-01-11 at 15 09 34


> I'd recommend reducing to 20 x 20.

I think you meant 24 x 24 for the gravatar size (at least that's the one used on the component on your screenshot).

> As far as linking to Gravatar this made me wonder if linking to their default blog wouldn't be better

Curious to know how do you feel about this @timmyc. Should we revert back to the default blog even if it's not consistent with the post like box.

**Comment by timmyc** [comment]
> Curious to know how do you feel about this @timmyc. Should we revert back to the default blog even if it's not consistent with the post like box.

I'm fine with either option to be honest.  We can always change it later.

**Comment by karmatosed** [comment]
As other links are to primary blog, can we go to that then if no issue?

**Comment by timmyc** [comment]
> As other links are to primary blog, can we go to that then if no issue?

👍  - I think the only thing we need to do is check for existence of primary blog, then maybe fallback to Gravatar.  Thinking of my dear mom who randomly likes my blog posts, but does not have a blog 😆

**Comment by karmatosed** [comment]
@timmyc excellent point about defaults. That makes sense and lets make sure your mum still has a link :)

**Comment by timmyc** [comment]
Tested this out again - I think once the remaining URL issue is taken care of, this is good to 🚢 - a rebase might be good too 👍

**Comment by youknowriad** [comment]
Link updated in 78c7dc4 :)

**Comment by timmyc** [comment]
:shipit:

**Comment by karmatosed** [comment]
This looks great. I have just one bit of feedback and it's about padding at the bottom. I almost feel compared to other sections we need to have more?

2017-01-12 at 16 21

**Comment by karmatosed** [comment]
Thanks @youknowriad - so great to see this completed.

**Pull request merged**